### ISSUE TITLE
Island MIssions GUI not updating upon completing quest

### ISSUE THREAD

**Issue by Nekosity12** [BG-Software-LLC/SuperiorSkyblock2#617]
### Minecraft's Version

Purpur 1.17.1-R0.1-SNAPSHOT

### Plugin's Version

DEVb316

### Describe the bug

When you complete a quest the GUI doesn't update, it just shows the quest as not completed and doesnt show the next quest. Restarting the server updates the GUI.

### To Reproduce

- Complete mission
- Check GUI
- See mission not completed
- Restart server
- See mission completed

### Additional Information

https://youtu.be/NykQayvGkl8
At 0:53 skip to 3:00 (Waiting on server to restart in those 2 minutes, nothing interesting :P)

**Comment by OmerBenGera** [comment]
Mind sharing with me your missions file?

**Comment by Skizzles** [comment]
Sent on Discord

**Comment by OmerBenGera** [comment]
Seems to work fine for me.
I completed the welcome mission and the "getting supplies" mission opened. Then I completed this mission, and a new mission was opened for me and it's displayed in the menu.
I tested it using latest dev build of SSB, on Purpur-1379

**Comment by Skizzles** [comment]
What I tested:
- 1379 and 1394 with the same issue occurring
- Without any other plugins and still had the issue occur
- Wiping the missions file and had the issue
- Clearing all mission jars and the issue still persisted

For some reason, it just won't update the GUI until a relog or restart

**Comment by OmerBenGera** [comment]
I have no clue, as all menus work the same, so if one menu is updating all menus should be updated.

**Comment by OmerBenGera** [comment]
Also, closing and re-opening the menus refresh them, so it makes no sense that a relog will help.

**Comment by Skizzles** [comment]
[22:02:06 INFO]: [SuperiorSkyblock2] [DEBUG] Action: Menu Click, Target: ISkizzlesI, Item: OAK_SAPLING, Slot: 0
[22:02:06 INFO]: [SuperiorSkyblock2] [DEBUG] Action: Reward Mission, Mission: gettingsupplies, Target: ISkizzlesI, Auto Reward: false, Force Reward: false
[22:02:06 INFO]: [SuperiorSkyblock2] [DEBUG] Action: Complete Mission, Player: ISkizzlesI, Mission: gettingsupplies
[22:02:06 INFO]: [SuperiorSkyblock2] [DEBUG] Action: Database Execute, Query: REPLACE INTO players_missions (`player`,`name`,`finish_count`) VALUES(?,?,?);
[22:02:06 INFO]: [SuperiorSkyblock2] [DEBUG] Action: Open Menu, Target: ISkizzlesI, Menu: menuPlayerMissions
[22:02:07 INFO]: Island | Successfully sent ISkizzlesI the title!
[22:02:07 INFO]: Slimefun 4> You have given ISkizzlesI 1 "Tiny Karma Crystal"
[22:02:07 INFO]: Slimefun 4> You have given ISkizzlesI 1 "Tiny Karma Crystal"

Some debug from the situation, still unsure the cause of this.

**Comment by Neeekss** [comment]
Same issue for me I try lastest dev build but nothing changed

**Comment by OmerBenGera** [comment]
> 
> 
> Same issue for me I try lastest dev build but nothing changed

Let me know how to replicate the issue, as it works fine for me

**Comment by SOLJAOVLS12** [comment]
Same issue also for me with the latest dev build

**Comment by OmerBenGera** [comment]
> 
> 
> Same issue also for me with the latest dev build

Let me know how to replicate the issue, as it works fine for me

**Comment by Nekosity12** [comment]
Downgraded to a version before the new database update (devb260 specifically) and the missions GUI worked fine. Was going to update slowly from there until I ran into the issue again but unfortunately there wasn't a list that allowed me to access versions that far back. Skizz believes it might have something to do with converting the database (which is why I tested so far back)

**Comment by OmerBenGera** [comment]
I dont see how it has anything to do with the database migration, but I will look into that.

**Comment by OmerBenGera** [comment]
Also, you can find all dev builds on https://hub.bg-software.com/

**Comment by OmerBenGera** [comment]
Also, you can find all dev builds on https://hub.bg-software.com/

**Comment by SOLJAOVLS12** [comment]
It was round about then when I first seen this happen

**Comment by Skizzles** [comment]
@OmerBenGera 

Tested versions, found the issue happens with build 300 and above. So seems something in build 300 caused this issue to start happening.

**Comment by Skizzles** [comment]
@Neeekss @SOLJAOVLS12 

Could you test with build 299 and see if the issue occurs for you?

**Comment by Skizzles** [comment]
Tested after Build 329 and this seems to be resolved. @SOLJAOVLS12 @Neeekss give it a go as well.

**Comment by OmerBenGera** [comment]
Let me know if it works for you as well.
Also, not sure if I have mentioned it yet, but I am working on a new missions system anyways, so it even if it's not fixed in these builds, it surely will be fixed when the new system comes out.

**Comment by SOLJAOVLS12** [comment]
Ok thank you will grab the jar now and test and let you no

**Comment by SOLJAOVLS12** [comment]
So i added b333 and same thing still happens with certain players

**Comment by SOLJAOVLS12** [comment]
Would removing there island and creating a new one also clear there missions tho as on my server I have a level system were to level up they have to complete certain levels so if it does they would have to redo them all

**Comment by Skizzles** [comment]
> So for example if I was to set my self as leader on players islands then they leave the island then I invite them back and set them as leader that should fix things right

Honestly I am unsure, it may update it or it may not.

**Comment by TheMartyCZ** [comment]
> > I don't know how they should be damaged, I tried a newer version .. it doesn't work so I'm on an older one again. There was no error in the console.
> 
> You must use latest dev build, otherwise I will not give you support.

But when I use the latest build ... missions don't work.

**Comment by OmerBenGera** [comment]
> > > I don't know how they should be damaged, I tried a newer version .. it doesn't work so I'm on an older one again. There was no error in the console.
> > 
> > 
> > You must use latest dev build, otherwise I will not give you support.
> 
> But when I use the latest build ... missions don't work.

This entire issue is about them not working. If you dont try things on latest dev build, how can I fix the issue?

**Comment by TheMartyCZ** [comment]
Should I use a broken build with players on the server?

**Comment by OmerBenGera** [comment]
> 
> 
> Should I use a broken build with players on the server?

You should test the latest build and try to find what causes the issue. I don't care on which server, you can test it on localhost, as I am doing.

**Comment by Skizzles** [comment]
I can confirm there are no issues anymore on all new islands. There are issues on islands created during that island leader issue at which changing the leader and setting it back fixed for me and Nekosity.

**Comment by TheMartyCZ** [comment]
> You should test the latest build and try to find what causes the issue. I don't care on which server, you can test it on localhost, as I am doing.
But this problem only happens with multiple players, because the problem does not occur with everyone and every time.

**Comment by OmerBenGera** [comment]
> But this problem only happens with multiple players, because the problem does not occur with everyone and every time.

We already know which islands are broken, as we have stated already.

**Comment by OmerBenGera** [comment]
@Nekosity12 @Neeekss @SOLJAOVLS12 Does it happen for you?

**Comment by SOLJAOVLS12** [comment]
What the island missions not completing and I will test today as I haven't tested yet since I have been getting over pneumonia but starting to feel better now so will test the changing island leader and adding them back today.

**Comment by OmerBenGera** [comment]
Any news regarding this?

**Comment by SOLJAOVLS12** [comment]
No I still get the problem were they dont complete even after I change leader then set it back

**Comment by OmerBenGera** [comment]
@SOLJAOVLS12 Try latest build, may have found the issue.

**Comment by SOLJAOVLS12** [comment]
Ok will test it when I'm home and let u no

**Comment by OmerBenGera** [comment]
I finally found what caused it, should be fixed in latest build.

**Comment by SOLJAOVLS12** [comment]
Nice but I couldn't use latest build last time I added it as it crashed my server on load due to chunks or summat or is that fixed now

**Comment by OmerBenGera** [comment]
> Nice but I couldn't use latest build last time I added it as it crashed my server on load due to chunks or summat or is that fixed now

I don't know what you talk about.
If there are issues with latest builds that crash your game, open a new issue and send there logs.

**Comment by SOLJAOVLS12** [comment]
There was many issues already posted about it will test it again and if same happens will send the errors

**Comment by OmerBenGera** [comment]
> There was many issues already posted about it will test it again and if same happens will send the errors

There are 11 issues opened, none of them related to crashes besides your one which is missing the logs that I requested here.
And as I said on that issue, if you're not gonna provide me with logs, I will close it as it's useless.

**Comment by SOLJAOVLS12** [comment]
Ye at the time there was tho plus many people that added that version had the same issue. Will check it shortly and get back to you

**Comment by OmerBenGera** [comment]
> Ye at the time there was tho plus many people that added that version had the same issue. Will check it shortly and get back to you

Well, no one has reported anything to me, I don't understand how you want the issues to get resolved on their own...
